### DISCUSSION TITLE
What meshopt_SimplifyErrorAbsolute does ?

### DISCUSSION THREAD

**Discussion by GiuseppeCassisi96** [zeux/meshoptimizer#708] [Q&A] [answered]
Hi,

I wonder what meshopt_SimplifyErrorAbsolute does. Does it allow meshopt_simplify to return the error as an absolute value instead of relative to the mesh extent?

**Comment by zeux** [comment, marked as answer]
Yes, as described in https://github.com/zeux/meshoptimizer/blob/master/src/meshoptimizer.h#L335. If the mesh is 50 units across (eg positions XYZ are from -10 to 40), an error 0.01 without using this flag is equivalent to error 0.5 when using this flag.

**Reply by GiuseppeCassisi96** [reply]
So, when I use this flag, is useless multiply the error with the value returned by meshopt_simplifyScale to get an absolute error:

```
outError *= meshopt_simplifyScale(&groupVertexBuffer[0].pos.x,
        groupVertexBuffer.size(), sizeof(MINERVA_VERTEX));  
```

**Reply by zeux** [reply]
Yes - note that both input and output error become absolute, so you would need to change the error threshold as well.

**Reply by GiuseppeCassisi96** [reply]
What do you means with error error threshold ? Is it the error threshold used to determine if a lod cluster can be rendered ?

**Reply by zeux** [reply]
The flag changes the treatment of both target_error and result_error in meshopt_simplify, so if you use it you need to change target_error as well.

**Reply by GiuseppeCassisi96** [reply]
Ok so I need to multiply the `target_error ` with the output of `meshopt_simplifyScale` :

```
targetError *= meshopt_simplifyScale(&groupVertexBuffer[0].pos.x,
        groupVertexBuffer.size(), sizeof(MINERVA_VERTEX));
```

**Reply by zeux** [reply]
Yes, if you want to specify the target error as relative. https://github.com/zeux/meshoptimizer/blob/b092eee9823cbc38ec8dd6107ff1dedcd3ac9bfa/demo/main.cpp#L646 example does this.

**Reply by GiuseppeCassisi96** [reply]
Dosen't it should transform the error to absolute instead of relative ?

**Reply by zeux** [reply]
If you know the target error in absolute units you can just use the flag. If you want to simplify the mesh with an error of 1% but use absolute flag, then you need to convert yourself.

**Reply by GiuseppeCassisi96** [reply]
Ok many thanks. I would like ask you another question. Why you resize the `meshlet_triangles` array in this way after the mesh splitting ?  

```
const meshopt_Meshlet& last = meshlets[meshlet_count - 1];

meshlet_vertices.resize(last.vertex_offset + last.vertex_count);
meshlet_triangles.resize(last.triangle_offset + ((last.triangle_count * 3 + 3) & ~3));
meshlets.resize(meshlet_count);
```
Thanks

**Reply by zeux** [reply]
This just keeps the triangles array aligned by 4 bytes (or by 4 indices since each index is 1 byte); technically `meshlet_triangles.resize(last_triangle_offset + last.triangle_count * 3);` would work as well but it's usually better to keep data 4-byte aligned if it's going to be read by GPU.
